utils/models.py

```diff
@@ -8,3 +8,122 @@ class UUIDModel(models.Model):
 
     class Meta:
         abstract = True
+
+
+class SerializableMixin(models.Model):
+    """
+    Mixin to add custom serialization for django models in order to get the desired tree of models to
+    the downloadable JSON form. It detects relationships automatically (many to many not yet fully
+    supported). Check for the example for more details about the structure.
+
+    Attributes need to be defined in the extending model:
+
+    - serialize_fields (required)
+        - tuple of dicts:
+            - name (required), name of the field or relation that's going to be added to the serialized object
+            - accessor (optional), function that is called when value of the field is resolved and it takes the
+              actual field value as argument
+
+    - serialize_root_key (required only for the root object)
+        - it's used as key ONLY for the root object (where serializing starts)
+        - keys for the other related objects are determined by the name of the relation
+
+    Example usage and output:
+
+    class Post(SerializableMixin):
+        serialize_fields = (
+            { "name": "title" },
+            { "name": "content" },
+            {
+                "name": "created_at",
+                "accessor": lambda x: x.strftime("%Y-%m-%d")
+            }
+            { "name": "comments" },
+        )
+        serialize_root_key = "POST_OF_THE_DAY"
```

Comment:
is this `serialize_root_key` just for some verbosity? don't understand the technical value behind it